dali/test/python/test_pipeline_debug.py

```diff
@@ -427,3 +427,43 @@ def test_seed_generation_base():
     pipe1 = seed_rn50_pipeline_base()
     pipe2 = seed_rn50_pipeline_base()
     compare_pipelines(pipe1, pipe2, 8, 10)
+
+
+@pipeline_def(batch_size=8, num_threads=3, device_id=0, seed=47, debug=True)
+def device_change_rn50_pipeline_base():
+    jpegs, labels = fn.readers.file(
+        file_root=file_root, shard_id=0, num_shards=2, random_shuffle=True)
+    images = fn.decoders.image(jpegs, output_type=types.RGB)
+
+    if device_change_rn50_pipeline_base.change:
+        images = images.gpu()
+
+    output = fn.random_resized_crop(images, size=(224, 224))
+    return labels, output
+
+
+@raises(RuntimeError, glob='Input * for operator * is on * but was on * when created.')
+def test_device_change():
+    pipe = device_change_rn50_pipeline_base()
+    pipe.build()
+    device_change_rn50_pipeline_base.change = True
+    pipe.run()
+    device_change_rn50_pipeline_base.change = False
+    pipe.run()
+
+
+@pipeline_def(batch_size=8, num_threads=3, device_id=0, seed=47, debug=True)
+def cpu_after_gpu_pipeline():
+    jpegs, labels = fn.readers.file(
+        file_root=file_root, shard_id=0, num_shards=2, random_shuffle=True)
+    images = fn.decoders.image(jpegs, output_type=types.RGB, device='mixed')
+
+    output = fn.random_resized_crop(images, size=(224, 224), device='cpu')
```

Comment:
What if we add:
```images = images.as_cpu()```?

Reply:
We can force it with `images.get().as_cpu()` and it will work.

Reply:
On the other hand, we can put any NumPy code in our pipeline so we should not block this. Let us leave it as it is.